validphys2/src/validphys/gridvalues.py

```diff
@@ -155,11 +166,24 @@ def evaluate_luminosity(
 
         # as in the second of Eq.(4) in arXiv:1607.01831
         res = sum(a*b for a,b in itertools.product(r1,r2))
+    elif channel == 'zlum':
+        u, ubar = 2, -2
+        d, dbar = -1, 1
+        res = (
+            _parton_pair_lumi_inner(pdf_set=pdf_set, n=n, mx=mx, x1=x1, x2=x2, i=u, j=ubar) +
+            _parton_pair_lumi_inner(pdf_set=pdf_set, n=n, mx=mx, x1=x1, x2=x2, i=d, j=dbar)
+        )
+    elif channel == 'wlum':
+        u, dbar = 2, -1
+        d, ubar = 1, -2
+        res = (
+            _parton_pair_lumi_inner(pdf_set=pdf_set, n=n, mx=mx, x1=x1, x2=x2, i=u, j=dbar) +
+            _parton_pair_lumi_inner(pdf_set=pdf_set, n=n, mx=mx, x1=x1, x2=x2, i=d, j=ubar)
+        )
```

Comment:
Do you want `u ub` x `d db` only or all  `u-like ub-like`? I think it makes more sense for a `zlum` to have all generations.

Reply:
I want these lumis now

Reply:
I wanted just to make sure it was not an oversight.

Reply:
Fair enough.